tests/phpunit/integration/Core/Util/User_Input_SettingsTest.php

```diff
@@ -0,0 +1,193 @@
+<?php
+/**
+ * Class Google\Site_Kit\Tests\Core\Util\User_Input_SettingsTest
+ *
+ * @package   Google\Site_Kit\Tests\Core\Util
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Core\Util;
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Storage\User_Options;
+use Google\Site_Kit\Core\Util\User_Input_Settings;
+use Google\Site_Kit\Tests\Core\Util\FakeUser_Input_Settings;
+use Google\Site_Kit\Tests\TestCase;
+
+class User_Input_SettingsTest extends TestCase {
+
+	/**
+	 * Context object.
+	 *
+	 * @var Context
+	 */
+	private $context;
+
+	/**
+	 * @var bool
+	 */
+	protected static $old_wp_using_ext_object_cache;
+
+	public static function wpSetUpBeforeClass() {
+		self::$old_wp_using_ext_object_cache = wp_using_ext_object_cache();
+	}
+
+	public static function wpTearDownAfterClass() {
+		wp_using_ext_object_cache( self::$old_wp_using_ext_object_cache );
+	}
+
+	public function setUp() {
+		parent::setUp();
+		$this->context = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+	}
+
+	public function test_not_connected_to_proxy() {
+		$settings = new User_Input_Settings( $this->context );
+		$results  = array(
+			$settings->get_settings(),
+			$settings->set_settings( array() ),
+		);
+
+		foreach ( $results as $result ) {
+			$this->assertTrue( is_wp_error( $result ) );
+			$this->assertEquals( 'not_connected', $result->get_error_code() );
+
+			$data = $result->get_error_data();
+			$this->assertArrayHasKey( 'status', $data );
+			$this->assertEquals( 400, $data['status'] );
+		}
+	}
+
+	public function test_get_settings_from_cache() {
+		wp_using_ext_object_cache( false );
+
+		$user_id = $this->factory()->user->create( array( 'role' => 'administrator' ) );
+		wp_set_current_user( $user_id );
+
+		$user_options = new User_Options( $this->context, $user_id );
+		$settings     = new FakeUser_Input_Settings( $this->context );
```

Comment:
Ideally we wouldn't use a fake instance of the class under test, but I suppose it's better than some alternatives here. If it wasn't `final`, I would prefer mocking `Authentication` but that's not an option right now.

Instead, let's mock `User_Input_Settings` in the test since it's not final, and stub out the `is_connected_to_proxy` method here, which will be more clear here rather than using a `FakeUser_Input_Settings` class.

Reply:
Updated.